### DISCUSSION TITLE
How can we view the API request body sent to a third party (Anthropic/OpenAI)?

### DISCUSSION THREAD

**Discussion by scpedicini** [danny-avila/LibreChat#4530] [Q&A]
I've configured LibreChat to work with Anthropic Sonnet 3.5 model but it has configuration values for Top P / Top K which Anthropic recommends *not* sending when passing a temperature value.

I set up a preset for Claude with some custom instructions / temperature but there doesn't seem to be a way to disable Top P or Top K. I also tried turning on `DEBUG_LOGGING` but the docker messages don't show the true payload that is being sent so its hard to know what is actually going on.

_EDIT: Looking at the logs it looks like it is sending top_p and top_k values unfortunately. (Assuming this represents the data being sent to the API)_

```
2024-10-23T21:39:03.073Z debug: [AnthropicClient]
{
  model: "claude-3-5-sonnet-20241022",
  stream: true,
  stop_sequences: undefined,
  temperature: 0.5,
  metadata.user_id: "158199ee0d641f70157e4",
  top_p: 0.7,
  top_k: 5,
  // 5 message(s)
  messages: [snip],
  max_tokens: 8192,
  // 1 system(s)
  system: [{"type":"text","text":"\nYou are Claude Coder","cache_control":{"type":"ephemeral"}}],
}
```

**Comment by THByte** [comment]
Hi @scpedicini, 
have you tried the `dropParams` configuration for the endpoint. It is mentioned [here](https://www.librechat.ai/docs/configuration/librechat_yaml/object_structure/custom_endpoint#dropparams)?
